src/wallet3/wallet.cpp

```diff
@@ -12,28 +12,32 @@
 #include <sqlitedb/database.hpp>
 #include <oxenmq/oxenmq.h>
 
-#include "common/fs.h"
 #include <future>
 #include <chrono>
 #include <thread>
 
 #include <iostream>
 
+#include <oxen/log.hpp>
+
+#include <spdlog/sinks/rotating_file_sink.h>
+
 namespace wallet
 {
+  static auto logcat = oxen::log::Cat("wallet");
+
   Wallet::Wallet(
       std::shared_ptr<oxenmq::OxenMQ> omq,
-      std::shared_ptr<Keyring> keys,
+      std::shared_ptr<Keyring> keyring,
       std::shared_ptr<TransactionConstructor> tx_constructor,
       std::shared_ptr<DaemonComms> daemon_comms,
       std::string_view dbFilename,
       std::string_view dbPassword,
       wallet::Config config_in)
       : omq(omq)
-      , db{std::make_shared<WalletDB>(fs::path(dbFilename), dbPassword)}
-      , keys{keys}
-      , tx_scanner{keys, db}
+      , db{std::make_shared<WalletDB>(file_path_from_default_datadir(config_in, dbFilename), dbPassword)}
       , tx_constructor{tx_constructor}
+      , tx_scanner{keyring, db}
       , daemon_comms{daemon_comms}
```

Comment:
Not added in this PR, but: all of these `shared_ptr` initializations should be `std::move()`d into place.